### PR TITLE
Fix block directory author average rating formatted as intereg

### LINKED ISSUE #24731: Block directory author average rating is formatted as an integer but the value returned by the REST API is a float
**Describe the bug**

When searching for a block from the block directory, the block author's average rating is returned by the REST API endpoint as a float. 

__Edit: Although in doing more research it does appear that the schema defines the rating as an integer. I'm not sure if that's a related or separate issue (or not an issue at all).__

**To reproduce**

1. Search for a block by an author with an average rating that isn't a whole number. As of the moment of this posting my block plugin, "Lite LMS Progress Tracker by LifterLMS" has a rating of 4.5 so it'll work.
2. The REST API response for the search which responds with the actual (non-whole number) average rating (with my plugin that's 4.5)
3. On screen 4 is displayed for the author's average rating. See screenshot below

**Expected behavior**

4.5 would be displayed on screen.

**Screenshots**

Screenshot_2020-08-21_14-56-01

**Editor version (please complete the following information):**
- WordPress version: 5.5
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default?gutenberg plugin
- If the Gutenberg plugin is installed, which version is it? 8.8.0 (from master branch)

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04.1
 - Browser: Brave
 - Version: Version 1.8.96 Chromium: 81.0.4044.138 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by thomasplevy**
## Description

Fixes #24731 

Uses `f` in favor of `d` to format the rating as a `float` in favor of an `int`.

## How has this been tested?

+ Manually tested against the recreation steps I noted in the related issue.
+ Locally ran `npm run test` to ensure this change didn't break any existing tests.

```
Test Suites: 408 passed, 408 total
Tests:       4376 passed, 4376 total
Snapshots:   198 passed, 198 total
Time:        39.582s
Ran all test suites.
```

+ I don't see any tests that specifically touch this particular area and I'm not sure what the best practice for creating a brand new test are so I haven't endeavored to write new tests. I'll be happy to update the PR if a regular contributor or reviewer would give me the go ahead to write new tests for the affected component. I've never written a test with snapshots before but I can figure it out.

## Screenshots 

Compare to initial screenshot submitted in bug report

Screenshot_2020-08-21_15-16-56

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-directory/src/components/downloadable-block-author-info/index.js [hunk 1]
```diff
@@ -23,8 +23,8 @@ function DownloadableBlockAuthorInfo( {
 					? sprintf(
 							/* translators: 1: number of blocks. 2: average rating. */
 							_n(
-								'This author has %1$d block, with an average rating of %2$d.',
-								'This author has %1$d blocks, with an average rating of %2$d.',
+								'This author has %1$d block, with an average rating of %2$f.',
+								'This author has %1$d blocks, with an average rating of %2$f.',
```
Comment (ryelle) [inline packages/block-directory/src/components/downloadable-block-author-info/index.js:27]:
```suggestion
								'This author has %1$d block, with an average rating of %2$.1f.',
								'This author has %1$d blocks, with an average rating of %2$.1f.',
```

**Review by ryelle — CHANGES_REQUESTED** [review]
Thanks for the PR, good catch. I've just got one suggestion, to limit it to one decimal place, otherwise ratings with ⅓ end up pretty long:

Screen Shot 2020-08-26 at 1 06 48 PM

File: packages/block-directory/src/components/downloadable-block-author-info/index.js (hunk 1 above)
Comment (thomasplevy) [reply packages/block-directory/src/components/downloadable-block-author-info/index.js:27]: Oh yes, I hadn't considered that, thank you!

Comment (thomasplevy) [reply packages/block-directory/src/components/downloadable-block-author-info/index.js]: Updated per notes above, thanks again

**Comment by thomasplevy** [comment]
@ryelle I also noticed that this will make something with a whole number average rating shows with a decimal. This is a change from previous behavior but I think it's probably an expected change that no one will question if they happen to notice the change, right?

Screenshot_2020-08-26_13-11-22

**Review by ryelle — APPROVED** [review]
Sorry it took me so long to get back to this issue! thanks again for working on it.

> I also noticed that this will make something with a whole number average rating shows with a decimal.

Yeah, I think that's fine.

**Comment by thomasplevy** [comment]
No worries about the wait and thanks for accepting.

I appreciate you pointing me in the right

**Pull request merged**